tests/CMakeLists.txt

```diff
@@ -6,5 +6,6 @@ foreach(TEST_SOURCE ${TESTS})
     string(REPLACE ".cpp" "" TEST_TARGET "${TEST_SOURCE}")
     add_executable(${TEST_TARGET} ${TEST_SOURCE})
     target_compile_features(${TEST_TARGET} PRIVATE cxx_std_17)
+    target_compile_definitions(${TEST_TARGET} PRIVATE CMAKE_SOURCE_DIR="${CMAKE_SOURCE_DIR}")
```

Comment:
it seems the tests require users to run them by `cmake -B` and `make -C`, ie from the root or maybe even from the tests directory? the tests use relative paths so if you try anything else like a typical workflow of going into the build dir to run them it will fail. ive instead passed the absolute dir into the tests with this define and then used it to make the test paths non-relative so no matter what way you run it it works